Tests/SentryTests/Helper/SentryAppStateManagerTests.swift

```diff
@@ -10,13 +10,14 @@ class SentryAppStateManagerTests: XCTestCase {
         let options: Options
         let fileManager: SentryFileManager
         let currentDate = TestCurrentDateProvider()
+        let dispatchQueue = TestSentryDispatchQueueWrapper()
 
         init() {
             options = Options()
             options.dsn = SentryAppStateManagerTests.dsnAsString
             options.releaseName = TestData.appState.releaseName
 
-            fileManager = try! SentryFileManager(options: options, andCurrentDateProvider: currentDate)
+            fileManager = try! SentryFileManager(options: options, andCurrentDateProvider: currentDate, dispatchQueueWrapper: dispatchQueue)
```

Comment:
So now in a bunch of test cases we can provide `TestSentryDispatchQueueWrapper()` so that the delayed dispatch to delete old files in SentryFileManager doesn't actually run, which was causing crashes due to a zombie (we may want to look into a possible retain cycle there).

Reply:
Yep, that makes sense. @kevinrenskers ⬆️ . I approved your PR adding this code. This is what we should have done to avoid the problem above.

Reply:
I misplaced some changes I made to `TestSentryDispatchQueueWrapper` that actually would make it so that delayed dispatches aren't executed; they're currently in this other PR: [control properties](https://github.com/getsentry/sentry-cocoa/pull/2228/files#diff-9fe39e2770ff3265a2cd31f007def3e23582062a213bc4bf959484ebc480c03cR8-R13) and [how they work](https://github.com/getsentry/sentry-cocoa/pull/2228/files#diff-9fe39e2770ff3265a2cd31f007def3e23582062a213bc4bf959484ebc480c03cR41-R47). I can move those changes into this PR instead.

Reply:
Done, see https://github.com/getsentry/sentry-cocoa/pull/2424/commits/671183f848fb36509e1b03e6e124530594d844f6